pkg/node/local_node_store.go

```diff
@@ -58,8 +59,11 @@ type LocalNodeStore struct {
 	// Changes to the local node are observable.
 	stream.Observable[LocalNode]
 
-	mu       lock.Mutex
+	mu    lock.Mutex
+	getMu lock.RWMutex
```

Comment:
Could you document the relationship between these two locks? Does holding the getMu imply holding the mu? In which order is it allowed to lock these locks, which order leads to deadlocks? What fields do each of these protect?